index.js

```diff
@@ -3,53 +3,187 @@ const state = {
+
+    const addToCartButton = document.createElement("button");
+    addToCartButton.innerText = "Add to cart";
+
+    // Call a function when the button is clicked that will add the item to cart
+    addToCartButton.addEventListener("click", (event) => {
+      addToCart(storeItem);
+    });
+
+    // Append the elements
+    storeItemsUL.append(itemLI);
+    itemLI.append(itemDIV);
+    itemDIV.append(itemImage);
+    itemLI.append(addToCartButton);
+  });
+}
+
+// Make an array to store the item
+
+function addToCart(storeItem) {
+  // Create a copy of the store item object
+  const storeItemCopy = { ...storeItem };
+
+  // Check if the cart already contains the item
+  // Increment the quantity if it contains the item
+  for (let i = 0; i < state.cart.length; i++) {
+    if (storeItemCopy.id === state.cart[i].id) {
+      state.cart[i].quantity++;
+      console.log("addToCart() same item: ", state.cart[i]);
+      updatePrice();
+      renderCartItems();
+      return null;
+    }
+  }
+  // If the item is not already in cart, then add to cart array
+  storeItemCopy.quantity = 1;
+  state.cart.push(storeItemCopy);
+  updatePrice();
+  renderCartItems();
+}
+
+function renderCartItems() {
+  cartUL.innerHTML = ""; // Throw away the HTML so that I can update the cart
+  // Create the necessary HTML elements & set attributes
+
+  console.log("Rendering: ", state.cart);
+
+  state.cart.forEach((cartItem) => {
+    const cartLI = document.createElement("li");
+
+    const cartImage = document.createElement("img");
+    cartImage.setAttribute("class", "cart--item-icon");
+    cartImage.setAttribute("src", `assets/icons/${cartItem.id}.svg`);
+    cartImage.setAttribute("alt", `${cartItem.name}`);
+
+    const cartText = document.createElement("p");
+    cartText.innerText = `${cartItem.name}`;
+
+    const decreaseQuantity = document.createElement("button");
+    decreaseQuantity.setAttribute("class", "quantity-btn remove-btn center");
+    decreaseQuantity.innerText = "-";
+
+    const quantityAmount = document.createElement("span");
+    quantityAmount.setAttribute("class", "quantity-text center");
+    quantityAmount.innerText = `${cartItem.quantity}`; // Change to the actual quantity amount!
+
+    const increaseQuantity = document.createElement("button");
+    increaseQuantity.setAttribute("class", "quantity-btn add-btn center");
+    increaseQuantity.innerText = "+";
+
+    // Click event to decrement quantity
+    decreaseQuantity.addEventListener("click", (event) => {
+      cartItem.quantity--;
+      if (cartItem.quantity === 0) {
+        state.cart.splice(state.cart.indexOf(cartItem), 1);
+      }
+      updatePrice();
+      renderCartItems();
+    });
+
+    // Click event to increment quantity
+    increaseQuantity.addEventListener("click", (event) => {
+      cartItem.quantity++;
+      updatePrice();
+      renderCartItems();
+    });
+
+    // Append the elements
+    cartUL.append(cartLI);
+    cartLI.append(cartImage);
```

Comment:
You can have multiple appends on a single line, like cartLI.append(cartImage, cartText... etc etc) just to save some space!